cdap-ui/app/cdap/components/ConnectionExample/index.js

```diff
@@ -0,0 +1,146 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, {Component} from 'react';
+import Datasource from '../../services/datasource';
+import Rx from 'rx';
+
+export default class ConnectionExample extends Component {
+  constructor(props) {
+    super(props);
+
+    this.dataSrc = new Datasource();
+
+    this.state = {
+      data: [],
+      error: '',
+      isPolling: true
+    };
+    let streamsReqObj = {
+      _cdapPath: '/namespaces/default/streams',
+      interval: 2000
+    };
+    let datasetReqObj = {
+      _cdapPath: '/namespaces/default/data/datasets'
+    };
+
+    let streams = this.dataSrc.poll(streamsReqObj)
+      .map((response) => {
+        return response.map((stream) => {
+          return {
+            name: stream.name,
+            type: 'Stream'
+          };
+        });
+      });
+    let datasets = this.dataSrc.poll(datasetReqObj)
+      .map((response) => {
+        return response.map((dataset) => {
+          return {
+            name: dataset.name,
+            type: 'Dataset'
+          };
+        });
+      });
+
+    let combinedData = Rx.Observable.combineLatest(streams, datasets);
+    this.combination = combinedData.subscribe(
+      (response) => {
+        let combined = response[0].concat(response[1]);
+        this.setState({data: combined});
+      },
+      (error) => {
+        // This error will get executed if either streams or datasets polling
+        // gives an error
+
+        this.setState({error: error});
+      });
+  }
+
+  addStream() {
+    let streamName = this.refs.streamName.value;
+
+    let reqObj = {
+      _cdapPath: '/namespaces/default/streams/' + streamName,
+      method: 'PUT'
+    };
+    this.dataSrc.request(reqObj)
+      .subscribe(
+        () => {
+          this.refs.streamName.value = '';
+        },
+        (err) => {
+          this.setState({error: err});
+        }
+      );
+  }
+
+  stopPoll() {
+    // To stop the poll, just dispose the subscription
+    this.combination.dispose();
+    this.setState({isPolling: false});
+  }
+
+  render() {
+    let error;
+    if (this.state.error) {
+      error = (
+        <div className="text-danger">
+          {this.state.error}
+        </div>
+      );
+    }
+
+    return (
+      <div>
+        <h1> Connection Example</h1>
+        <h3> Datasets & Streams <small>in Default Namespace</small></h3>
+
+        {error}
+        <div className="row">
+          <div className="col-xs-3">
+            <input type="text"
```

Comment:
The general practice is to use refs in some callbacks need to be executed to communicate to children or from children. String for refs is an anti-pattern and we should probably avoid it. 

Here the `defaultValue` or `value` can be used and we need to maintain it with a state.

Reference: https://facebook.github.io/react/docs/more-about-refs.html#the-ref-string-attribute